qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -0,0 +1,338 @@
+    is needed. At the minimum the developer must set the class variable :code:`__updater__` which
+    should have an :code:`update` method and can be chosen from the library
+    :mod:`qiskit_experiments.calibration_management.update_library`. See also
+    :class:`qiskit_experiments.calibration_management.update_library.BaseUpdater`. If no updater
+    is specified the experiment will still run but no update of the calibrations will be performed.
+
+    In addition to the calibration specific requirements, the developer must set the analysis method
+    with the class variable :code:`__analysis_class__` and any default experiment options.
+    """
+
+    # The updater class that updates the Calibrations instance. Different calibration
+    # experiments will use different updaters.
+    __updater__ = None
+
+    # pylint: disable=super-init-not-called
+    def __init__(
+        self,
+        calibrations: Union[BackendCalibrations, Calibrations],
+        schedule_name: Optional[str] = None,
+        cal_parameter_name: Optional[str] = None,
+        auto_update: Optional[bool] = True,
+    ):
+        """Setup the calibration experiment object.
+
+        Args:
+            calibrations: The calibrations instance with which to initialize the experiment.
+            schedule_name: An optional string which specifies the name of the schedule in
+                the calibrations that will be updated.
+            cal_parameter_name: An optional string which specifies the name of the parameter in
+                the calibrations that will be updated. If None is given then no parameter will
+                be updated. Subclasses may assign default values in their init.
+            auto_update: If set to True (the default) then the calibrations will automatically be
+                updated once the experiment has run and :meth:`block_for_results()` will be called.
+        """
+        self._cals = calibrations
+        self._sched_name = schedule_name
+        self._param_name = cal_parameter_name
+        self.auto_update = auto_update
+
+    def update_calibrations(self, experiment_data: ExperimentData):
+        """Update parameter values in the :class:`Calibrations` instance.
+
+        The default behaviour is to call the update method of the class variable
+        :code:`__updater__` with simplistic options. Subclasses can override this
+        method to update the instance of :class:`Calibrations` if they require a
+        more sophisticated behaviour as is the case for the :class:`Rabi` and
+        :class:`FineAmplitude` calibration experiments.
+        """
+        if self.__updater__ is not None:
+            self.__updater__.update(
+                self._cals,
+                experiment_data,
+                parameter=self._param_name,
+                schedule=self._sched_name,
+            )
+
+    def _get_schedule_from_options(self, option_name: str) -> ScheduleBlock:
```

Comment:
This could have dual purpose. One will take complete pulse program from options, and other can directly generate a program with parameters in options on the fly. How do we solve this ambiguity? Seems like this comes down to discussion of flexibility (of pulse shape) v.s. simplicity (single liner execution).

Reply:
I think this is up to the developer to decide how he wants to implement this method.